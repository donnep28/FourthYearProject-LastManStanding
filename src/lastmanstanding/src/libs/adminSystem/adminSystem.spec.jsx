import React from 'react';
import {shallow} from 'enzyme';
import AdminSystem from './adminSystem';
import axios from 'axios';

const mockSetState = jest.fn();

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: (initial) => [initial, mockSetState],
}));

jest.mock('axios', () => ({
  post: jest.fn((_url, _body) => {
    return new Promise((resolve) => {
      url = '';
      body = {};
      resolve(true);
    });
  }),
}));

describe('leagues', () => {
  beforeAll(() => {
    jest.resetAllMocks();
  });

  describe('Snapshot', () => {
    it('renders player removal', () => {
      const props = {
        leagueStatus: true,
        leagueID: 'TestID',
        playerRemoval: true,
        leaguePlayerID: 'TestplayerID',
        setRender: jest.fn(),
      };
      const component = shallow(
          <AdminSystem {...props}/>,
      );
      expect(component).toMatchSnapshot();
    });
    it('renders admin tools with league locked', () => {
      const props = {
        leagueStatus: true,
        leagueID: 'TestID',
        playerRemoval: false,
        leaguePlayerID: 'TestplayerID',
        setRender: jest.fn(),
      };
      const component = shallow(
          <AdminSystem {...props}/>,
      );
      expect(component).toMatchSnapshot();
    });
    it('renders admin tools with league unlocked', () => {
      const props = {
        leagueStatus: false,
        leagueID: 'TestID',
        playerRemoval: false,
        leaguePlayerID: 'TestplayerID',
        setRender: jest.fn(),
      };
      const component = shallow(
          <AdminSystem {...props}/>,
      );
      expect(component).toMatchSnapshot();
    });
  });
  describe('Button Clicks', () => {
    it('unlock league', async () => {
      const props = {
        leagueStatus: true,
        leagueID: 'TestID',
        playerRemoval: false,
        leaguePlayerID: 'TestplayerID',
        setRender: jest.fn(),
      };
      const component = shallow(
          <AdminSystem {...props}/>,
      );
      const response = 'Successfully unlocked League';
      global.confirm = () => true;
      axios.post.mockImplementationOnce(() => Promise.resolve(response));
      component.find('[data-automation="unlockLeague"]').simulate('click');
      expect(axios.post).toHaveBeenCalled();
    });
    it('lock league', async () => {
      const props = {
        leagueStatus: false,
        leagueID: 'TestID',
        playerRemoval: false,
        leaguePlayerID: 'TestplayerID',
        setRender: jest.fn(),
      };
      const component = shallow(
          <AdminSystem {...props}/>,
      );
      const response = 'Successfully locked League';
      global.confirm = () => true;
      axios.post.mockImplementationOnce(() => Promise.resolve(response));
      component.find('[data-automation="lockLeague"]').simulate('click');
      expect(axios.post).toHaveBeenCalled();
    });
  });
});
