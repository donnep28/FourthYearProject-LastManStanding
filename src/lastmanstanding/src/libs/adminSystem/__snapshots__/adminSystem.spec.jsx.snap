// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`leagues Snapshot renders admin tools with league locked 1`] = `
<div>
  <WithStyles(ForwardRef(Tooltip))
    placement="top"
    title="Unlock League"
  >
    <WithStyles(ForwardRef(Button))
      data-automation="unlockLeague"
      onClick={[Function]}
    >
      <Memo(ForwardRef(LockIcon))
        style={
          Object {
            "color": "white",
          }
        }
      />
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Tooltip))>
  <WithStyles(ForwardRef(Tooltip))
    placement="top"
    title="Reset League"
  >
    <WithStyles(ForwardRef(Button))
      onClick={[Function]}
    >
      <Memo(ForwardRef(RotateLeftIcon))
        style={
          Object {
            "color": "#ffffff",
          }
        }
      />
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Tooltip))>
  <WithStyles(ForwardRef(Tooltip))
    placement="top"
    title="Delete League"
  >
    <WithStyles(ForwardRef(Button))
      onClick={[Function]}
    >
      <Memo(ForwardRef(DeleteIcon))
        style={
          Object {
            "color": "#ffffff",
          }
        }
      />
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Tooltip))>
</div>
`;

exports[`leagues Snapshot renders admin tools with league unlocked 1`] = `
<div>
  <WithStyles(ForwardRef(Tooltip))
    placement="top"
    title="Lock League"
  >
    <WithStyles(ForwardRef(Button))
      data-automation="lockLeague"
      onClick={[Function]}
    >
      <Memo(ForwardRef(LockOpenIcon))
        style={
          Object {
            "color": "white",
          }
        }
      />
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Tooltip))>
  <WithStyles(ForwardRef(Tooltip))
    placement="top"
    title="Reset League"
  >
    <WithStyles(ForwardRef(Button))
      onClick={[Function]}
    >
      <Memo(ForwardRef(RotateLeftIcon))
        style={
          Object {
            "color": "#ffffff",
          }
        }
      />
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Tooltip))>
  <WithStyles(ForwardRef(Tooltip))
    placement="top"
    title="Delete League"
  >
    <WithStyles(ForwardRef(Button))
      onClick={[Function]}
    >
      <Memo(ForwardRef(DeleteIcon))
        style={
          Object {
            "color": "#ffffff",
          }
        }
      />
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Tooltip))>
</div>
`;

exports[`leagues Snapshot renders player removal 1`] = `
<WithStyles(ForwardRef(IconButton))
  data-automation="removePlayer"
  onClick={[Function]}
>
  <Memo(ForwardRef(DeleteIcon))
    style={
      Object {
        "color": "red",
      }
    }
  />
</WithStyles(ForwardRef(IconButton))>
`;
